ui/src/reducers/index.js

```diff
@@ -35,6 +29,29 @@ export default (state = initialState, action) => {
         },
         apps: sortedApps
       };
+    case APPS_REQUEST:
+      return {
+        ...state
+      };
+    case APPS_SUCCESS:
+      var fetchedApps = [];
+      for (var i = 0; i < action.result.length; i++) {
+        var temp = [];
+        temp[0] = action.result[i].appId;
```

Comment:
This might not be the best way to do this. It's a bit rough and ready still.

Reply:
You should be able to replace the `for` loop with `map()` on the array if you wanted, but the explicit code looks fine to me

Reply:
yeah I used my java knowledge here, I knew it wasn't ideal. Going to look into `map()` with js.

Reply:
tbh, may not really be too many gains there, but might be slightly more readable.